scan/enum.go

```diff
@@ -0,0 +1,100 @@
+package scan
+
+import (
+	"go/ast"
+	"strconv"
+	"unicode"
+	"strings"
+)
+
+func enumName(comments *ast.CommentGroup, typeSpec *ast.TypeSpec) (name string, isEnum bool) {
+	if comments != nil {
+		for _, cmt := range comments.List {
+			for _, ln := range strings.Split(cmt.Text, "\n") {
+				matches := rxEnumOverride.FindStringSubmatch(ln)
+				if len(matches) > 1 {
+					name = matches[1]
+					if name == "" {
+						return typeSpec.Name.String(), true
+					}
+					return strings.TrimSpace(name), true
+				}
+			}
+		}
+	}
+	return "", false
+}
+
+func upperSnakeCase(s string) string {
```

Comment:
Sorry, I am new starter of Golang.
It may be hard for me to manage codes cross deps.
This why I have not write testing codes too. 😿